### PR TITLE
Support for VarScan2 Module

### PR DESCRIPTION
**Pull request by apeltzer**
This adds support for VarScan2, allowing to read all types of reports from VarScan2 properly as tested with the test data.

## If this PR is for a new module
 - [x] There is example tool output for tools in the https://github.com/ewels/MultiQC_TestData repository
 - [x] Code is tested and works locally (including with `--lint` flag)
 - [x] `CHANGELOG.md` is updated
 - [x] `README.md` is updated
 - [x] `docs/README.md` is updated with link to below
 - [x] `docs/modulename.md` is created
 - [x] Everything that can be represented with a plot instead of a table is a plot
 - [x] Report sections have a description and help text (with `self.add_section`)
 - [x] There aren't any huge tables with > 6 columns (explain reasoning if so)
 - [x] Each table column has a different colour scale to its neighbour, which relates to the data (eg. if high numbers are bad, they're red)
 - [x] Module does not do any significant computational work

### CODE REVIEW

**Comment by apeltzer** [comment]
Exemplary MultiQC report on the multiqc testdata:

[multiqc_report_Varscan2.html.txt](https://github.com/ewels/MultiQC/files/4497235/multiqc_report_Varscan2.html.txt)

Linting passes, too:

```bash
--------------------------------
 MultiQC installation complete!
--------------------------------
For help in running MultiQC, please see the documentation available
at http://multiqc.info or run: multiqc --help

[WARNING]    lint_helpers : Can't check docs readme in lint test as file doesn't exist: /Users/alexanderpeltzer/.conda/envs/multiqc-dev/lib/python3.7/site-packages/multiqc-1.9.dev0-py3.7.egg/docs/README.md
[INFO   ]         multiqc : This is MultiQC v1.9.dev0
[INFO   ]         multiqc : Template    : default
[INFO   ]         multiqc : --lint specified. Being strict with validation.
[INFO   ]         multiqc : Searching   : /Users/alexanderpeltzer/IDEA/MultiQC_TestData/data/modules/varscan2
[INFO   ]         multiqc : Only using modules varscan2
[INFO   ]        varscan2 : Found 3 reports
[INFO   ]         multiqc : Compressing plot data
[WARNING]         multiqc : Deleting    : multiqc_report.html   (-f was specified)
[WARNING]         multiqc : Deleting    : multiqc_data   (-f was specified)
[INFO   ]         multiqc : Report      : multiqc_report.html
[INFO   ]         multiqc : Data        : multiqc_data
[INFO   ]         multiqc : MultiQC complete
```

**Comment by apeltzer** [comment]
Feedback from @JoseEspinosa and @drpatelh would be super cool :-) Also with respect to hiding columns, potential plotting ideas etc pp - would be easy to adjust/add.

File: multiqc/modules/varscan2/varscan2.py [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+            'title': 'Minimum coverage',
+            'description': 'Minimum coverage at variant position observed',
+            'min': 0,
+            'scale': 'RdYlGn-rev',
+            'hidden': True,
+            'format': '{:,.0f}'
+        }
+        headers['min_reads2'] = {
+            'title': 'Minimum reads2',
+            'description': 'Whatever this means',
```
Comment (ewels) [inline multiqc/modules/varscan2/varscan2.py:134]: 😆

**Comment by ewels** [comment]
Ok, I've had a play with the code and done a bit of refactoring. Notable, compressing the two bar graphs into one and removing / hiding most of the table columns.

[multiqc_report.html.zip](https://github.com/ewels/MultiQC/files/4559763/multiqc_report.html.zip)

Please have a look and let me know what you think @drpatelh / @apeltzer /  everyone!

Phil

**Comment by drpatelh** [comment]
Looks great! Thank you 🙂 

Slight comment. Can you make the colours in the plot "green" or "red" depending on pass/fail? Will stick out more visually.

**Comment by apeltzer** [comment]
I adjusted the colours as per Harshils comment :-)

**Comment by apeltzer** [comment]
(failed colours reddish, reported greenish - twice different reds /greens to be distinguishable)

**Comment by ewels** [comment]
I couldn't handle your colour choices sorry @apeltzer 😅 So I switched them with some taken from ColourBrewer. I made them the same for the two categories too, as I think reported / not reported is kind of the same. I made the legends change though to make it more obvious that the type of variant is changing.

Hope this is ok!

Phil

**Pull request merged**